roachpb/data.go

```diff
@@ -446,9 +446,9 @@ func (v *Value) SetDuration(t duration.Duration) error {
 // SetDecimal encodes the specified decimal value into the bytes field of
 // the receiver using Gob encoding, sets the tag and clears the checksum.
 func (v *Value) SetDecimal(dec *inf.Dec) error {
-	decSize := encoding.UpperBoundDecimalSize(dec)
-	v.RawBytes = make([]byte, headerSize, headerSize+decSize)
-	v.RawBytes = encoding.EncodeDecimalAscending(v.RawBytes, dec)
+	encodingSizeOverestimate := headerSize + encoding.OverestimateDecimalSize(dec)
+	v.RawBytes = make([]byte, encodingSizeOverestimate)
+	v.RawBytes = encoding.EncodeDecimalAscending(v.RawBytes[:headerSize], dec)
```

Comment:
Yeah good point. I added back a `EncodeNonSortingDecimal` function to use for `roachpb.Value`. Because this is not for key encoding, there were a few places I could gain some speed. They included cutting out the length/terminator from the encoding itself because the length of the `Value.dataBytes` is already known.